### DISCUSSION TITLE
"Unsupported integer type" for unsigned long long (uint64_t) value type of cuco::static_map

### DISCUSSION THREAD

**Discussion by Gibyeng** [NVIDIA/cuCollections#655] [Q&A] [answered]
When I create a static_map with unsigned long long type, there is a error: `fast_int.cuh(36): error: static assertion failed with "Unsupported integer type"`.
It seems that the reason is that current `INT128` is not supported. how to enable `INT128` setting?

**Comment by Gibyeng** [comment]
My nvcc version is 13.8 which should support `int128` and I test the following code in my machine 
``` c++
#ifdef __CUDACC__
  printf("__CUDACC__ defined\n");
#endif

#ifdef __SIZEOF_INT128__
  printf("__SIZEOF_INT128__ is defined: %d bytes\n", __SIZEOF_INT128__);
#else
  // Try a test for __int128 availability
  __int128 test = 1; // If this compiles, __int128 is supported.
  printf("__SIZEOF_INT128__ is not defined, but __int128 is supported.\n");
#endif
  return 0;
```
The output suggests that `Int128` is defined.
```
__CUDACC__ defined
__SIZEOF_INT128__ is defined: 16 bytes
```
it seems the compiler thinks `unsigned long long` is not the same type with `uint64_t`,  I used `uint64_t` instead then the error is gone.

**Reply by PointKernel** [reply, marked as answer]
The size of `unsigned long long` is architecture-dependent and, therefore, not guaranteed to match `uint64_t`. We are tracking the addition of `INT128` support for Hopper GPUs and newer architectures in #547. Please refer to it for updates on this feature request.
